elements/pf-select/pf-select-list.ts

```diff
@@ -0,0 +1,181 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import type { PropertyValueMap, PropertyValues } from 'lit';
+import { ListboxController, type ListboxOptionElement, type ListboxValue } from '@patternfly/pfe-core/controllers/listbox-controller.js';
+import { PfSelectGroup } from './pf-select-group.js';
+import { PfSelectOption } from './pf-select-option.js';
+
+import styles from './pf-select-list.css';
+/**
+ * Listbox for select options.
+ *
+ * @see {@link https://www.w3.org/WAI/ARIA/apg/patterns/listbox/|WAI-ARIA Listbox Pattern}
+ *
+ * @slot - insert `pf-select-option` and/or `pf-select-groups` here
+ */
+@customElement('pf-select-list')
+export class PfSelectList extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * whether listbox is disabled
+   */
+  @property({ reflect: true, attribute: 'disabled', type: Boolean }) disabled = false;
+
+  /**
+   * whether filtering (if enabled) will be case-sensitive
+   */
+  @property({ reflect: true, attribute: 'case-sensitive', type: Boolean }) caseSensitive = false;
+
+  /**
+   * whether option filtering is disabled
+   */
+  @property({ reflect: true, attribute: 'disable-filter', type: Boolean }) disableFilter = false;
+
+  /**
+   * whether filtering (if enabled) will look for filter match anywhere in option text
+   * (by default it will only match if option starts with filter)
+   */
+  @property({ reflect: true, attribute: 'match-anywhere', type: Boolean }) matchAnywhere = false;
+
+  /**
+   * whether multiple items can be selected
+   */
+  @property({ reflect: true, attribute: 'multi-selectable', type: Boolean }) multiSelectable = false;
+
+  #listboxController?: ListboxController;
+
+  /**
+   * filter string for visible options
+   */
+  set filter(filterText: string) {
+    if (this.#listboxController) {
+      this.#listboxController.filter = filterText;
+    }
+  }
+
+  get filter() {
+    return this.#listboxController?.filter || '';
+  }
+
+  /**
+   * array of slotted options
+   */
+  get options() {
+    const slotted = this.querySelector('slot')?.assignedElements() || [...this.querySelectorAll('pf-select-option')];
+    const options = slotted?.map(element => {
+      if (element instanceof PfSelectOption) {
+        return element;
+      } else if (element instanceof PfSelectGroup) {
+        return [...element.querySelectorAll('pf-select-option')];
+      }
+    }).flat();
+    return options;
+  }
```

Comment:
Maybe a refactor here?  

first including 

```
import { queryAssignedElements } from 'lit/decorators/query-assigned-elements.js';
...
@queryAssignedElements({ flatten: true }) slottedElements?: Array<HTMLElement>;`
```


```Suggestion
  /**
   * array of slotted options
   */
  get options() {
    if (!this.slottedElements || this.slottedElements.length === 0) {
      return;
    }

    const options = this.slottedElements.flatMap((element: HTMLElement) => {
      if (element instanceof PfSelectOption) {
        return element;
      } else if (element instanceof PfSelectGroup) {
        return [...element.querySelectorAll('pf-select-option')];
      }
    });
    return options;
  }
```

Reply:
Might refactor further too by adding a way to access options in a `PfSelectGroup`:

something like psuedo code:

```
      } else if (element instanceof PfSelectGroup) {
        return element.selectOptions;
      }
```